File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.java
class Solution {
    public int minElement(int[] nums) {
        int mini = Integer.MAX_VALUE;

        for (int num : nums){
            int current = 0;

            while (num > 0){
                current += num % 10;
                num /= 10;
            }

            mini = Math.min(mini, current);
        }

        return mini;
    }
}

File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.py
class Solution:
    def minElement(self, nums: List[int]) -> int:
        def sumDigit(num):
            total = 0

            while num > 0:
                total += num % 10
                num = num // 10

            return total

        mini = float('inf')

        for num in nums:
            mini = min(mini, sumDigit(num))

        return mini